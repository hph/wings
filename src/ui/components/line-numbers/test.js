import React from 'react';
import renderer from 'react-test-renderer';
import _ from 'lodash';

import LineNumbers from './index';

const defaultProps = {
  className: '',
  config: {
    relativeLineNumbers: false,
  },
  view: {
    lines: [],
    row: 0,
    firstVisibleRow: 0,
    firstVisibleColumn: 0,
  },
};

const createSnapshot = (props = defaultProps) => {
  expect(
    renderer.create(<LineNumbers {...props} />).toJSON(),
  ).toMatchSnapshot();
};

test('LineNumbers renders at least one line even when empty', () => {
  createSnapshot();
});

test('LineNumbers allows passing in an external classname', () => {
  createSnapshot({
    ...defaultProps,
    className: 'custom-classname',
  });
});

test('LineNumbers adds an overlay class when firstVisibleColumn is larger than 0', () => {
  createSnapshot({
    ...defaultProps,
    view: {
      ...defaultProps.view,
      firstVisibleColumn: 1,
    },
  });
});

test('LineNumbers renders line numbers', () => {
  createSnapshot({
    ...defaultProps,
    view: {
      ...defaultProps.view,
      lines: _.range(3),
    },
  });
});

test('LineNumbers starts zero-indexed when using relative line numbers', () => {
  createSnapshot({
    ...defaultProps,
    view: {
      ...defaultProps.view,
      lines: _.range(3),
    },
    config: {
      ...defaultProps.config,
      relativeLineNumbers: true,
    },
  });
});

test('LineNumbers renders relative line numbers when so configured', () => {
  createSnapshot({
    ...defaultProps,
    view: {
      ...defaultProps.view,
      lines: _.range(5),
      row: 2,
    },
    config: {
      ...defaultProps.config,
      relativeLineNumbers: true,
    },
  });
});

test('LineNumbers starts off at the first visible row', () => {
  createSnapshot({
    ...defaultProps,
    view: {
      ...defaultProps.view,
      lines: _.range(5),
      row: 2,
      firstVisibleRow: 2,
    },
  });
});

test('LineNumbers starts off at the first visible row with relative line numbers', () => {
  createSnapshot({
    ...defaultProps,
    view: {
      ...defaultProps.view,
      lines: _.range(5),
      row: 2,
      firstVisibleRow: 2,
    },
    config: {
      ...defaultProps.config,
      relativeLineNumbers: true,
    },
  });
});
