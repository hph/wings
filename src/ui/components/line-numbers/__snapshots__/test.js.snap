// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineNumbers adds an overlay class when firstVisibleChar is larger than 0 1`] = `
<div
  className="root overlay"
>
  <div>
    1
  </div>
</div>
`;

exports[`LineNumbers allows passing in an external classname 1`] = `
<div
  className="root custom-classname"
>
  <div>
    1
  </div>
</div>
`;

exports[`LineNumbers renders at least one line even when empty 1`] = `
<div
  className="root"
>
  <div>
    1
  </div>
</div>
`;

exports[`LineNumbers renders line numbers 1`] = `
<div
  className="root"
>
  <div>
    1
  </div>
  <div>
    2
  </div>
  <div>
    3
  </div>
</div>
`;

exports[`LineNumbers renders relative line numbers when so configured 1`] = `
<div
  className="root"
>
  <div>
    2
  </div>
  <div>
    1
  </div>
  <div>
    0
  </div>
  <div>
    1
  </div>
  <div>
    2
  </div>
</div>
`;

exports[`LineNumbers should only render at most as many lines as are visible in the viewport 1`] = `
<div
  className="root"
>
  <div>
    1
  </div>
  <div>
    2
  </div>
  <div>
    3
  </div>
  <div>
    4
  </div>
  <div>
    5
  </div>
  <div>
    6
  </div>
  <div>
    7
  </div>
  <div>
    8
  </div>
  <div>
    9
  </div>
  <div>
    10
  </div>
</div>
`;

exports[`LineNumbers starts off at the first visible line 1`] = `
<div
  className="root"
>
  <div>
    3
  </div>
  <div>
    4
  </div>
  <div>
    5
  </div>
</div>
`;

exports[`LineNumbers starts off at the first visible line with relative line numbers 1`] = `
<div
  className="root"
>
  <div>
    0
  </div>
  <div>
    1
  </div>
  <div>
    2
  </div>
</div>
`;

exports[`LineNumbers starts zero-indexed when using relative line numbers 1`] = `
<div
  className="root"
>
  <div>
    0
  </div>
  <div>
    1
  </div>
  <div>
    2
  </div>
</div>
`;
